src/libsyntax_pos/symbol.rs

```diff
@@ -261,73 +261,77 @@ macro_rules! declare_keywords {(
 declare_keywords! {
     // Special reserved identifiers used internally for elided lifetimes,
     // unnamed method parameters, crate root module, error recovery etc.
-    (0,  Invalid,        "")
-    (1,  CrateRoot,      "{{root}}")
-    (2,  DollarCrate,    "$crate")
+    (0,  Invalid,            "")
+    (1,  CrateRoot,          "{{root}}")
+    (2,  DollarCrate,        "$crate")
+    (3,  Underscore,         "_")
```

Comment:
I can move it into the keyword list below, the only difference would be whether error messages say "keyword `_`" or "reserved identifier `_`".

Reply:
I think "reserved identifier" is better than "keyword" as it doesn't really look/feel like a keyword.